Do not process the same finished job twice

milljm opened the pull request:
When a long running alert occurs, it was possible for the job to finish before
the alert was processed in the status pool. This caused job_count discrepancies
(and thus the exception) when deducting the amount of finished jobs the status
pool was processing.

This PR eliminates that issue, as well as using a set to determine if all jobs
submitted have been processed.

Closes #11932

Commit:
Do not process the same finished job twice

When a long running alert occurs, it was possible for the job to finish before
the alert was processed in the status pool. This caused job_count discrepancies
(and thus the exception) when deducting the amount of finished jobs the status
pool was processing.

This PR eliminates that issue, as well as using a set to determine if all jobs
submitted have been processed.

Closes #11932

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/216679/) on e135182 wanted to post the following:

View the site [here](http://mooseframework.inl.gov/docs/PRs/11933/site)

This comment will be updated on new commits.

Commit:
Add a new set for all jobs scheduled

Add a new set to contain all jobs submitted to the Scheduler.
Cleanup the jobStatus method a bit more.

Commit:
Change variable name

Makes more sense if I use the same name as specified in the comment!

permcody commented:
https://www.youtube.com/watch?v=D01mdGW9-Zg
